test/files/neg/t9847.check

```diff
@@ -0,0 +1,45 @@
+t9847.scala:4: warning: discarded non-Unit value
+  def f(): Unit = 42
+                  ^
+t9847.scala:4: warning: a pure expression does nothing in statement position
+  def f(): Unit = 42
+                  ^
```

Comment:
AbstractReporter suppresses multiple ERROR or lower-priority messages (warnings in presence of errors).

The value discard is a flag:

```
$ scala -Ywarn-value-discard
Welcome to Scala 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_92).
Type in expressions for evaluation. Or try :help.

scala> def f: Unit = 42
<console>:11: warning: discarded non-Unit value
       def f: Unit = 42
                     ^
<console>:11: warning: a pure expression does nothing in statement position; you may be omitting necessary parentheses
       def f: Unit = 42
                     ^
f: Unit

scala> :quit
$ scalam -Ywarn-value-discard
Welcome to Scala 2.12.0-M5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_92).
Type in expressions for evaluation. Or try :help.

scala> def f: Unit = 42
<console>:11: warning: discarded non-Unit value
       def f: Unit = 42
                     ^
<console>:11: warning: a pure expression does nothing in statement position; you may be omitting necessary parentheses
       def f: Unit = 42
                     ^
f: Unit

scala> :quit
$ skala -Ywarn-value-discard
Welcome to Scala 2.12.0-20160708-085210-f805cf5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_92).
Type in expressions for evaluation. Or try :help.

scala> def f: Unit = 42
<console>:11: warning: discarded non-Unit value
       def f: Unit = 42
                     ^
<console>:11: warning: a pure expression does nothing in statement position
       def f: Unit = 42
                     ^
f: Unit

scala> :quit

```